src/classes/__tests__/Gaussian.functions.test.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  ROOT_2LN2,
+  ROOT_PI_OVER_LN2,
+  GAUSSIAN_EXP_FACTOR,
+} from '../../util/constants';
+import erfinv from '../../util/erfinv';
+import * as gaussian from '../Gaussian';
+
+describe('Gaussian.shape', () => {
+  it('height 1', () => {
+    let data = gaussian.getData({ fwhm: 10, height: 1 });
+    expect(data).toHaveLength(39);
+    const center = (data.length - 1) / 2;
+    expect(data[center]).toBe(1);
+    let area = data.reduce((a, b) => a + b, 0);
+    expect(area).toBeCloseTo((ROOT_PI_OVER_LN2 * 10) / 2, 3);
+  });
+
+  it('check gaussian continuous', () => {
+    let y = gaussian.getData({ fwhm: 5900, factor: 1 });
+    const nbChanges = getNbChanges(y);
+    expect(nbChanges).toBe(2);
+  });
+
+  it('fwhm fixed and normalized', () => {
+    const fwhm = 50;
+    let data = gaussian.getData({ fwhm });
+    expect(data).toHaveLength(195);
+    let area = data.reduce((a, b) => a + b, 0);
+    expect(area).toBeCloseTo(0.9999, 2);
+    let height = Math.sqrt(-GAUSSIAN_EXP_FACTOR / Math.PI) / fwhm;
+    let computedArea = gaussian.getArea({ fwhm, height });
+    expect(computedArea).toBeCloseTo(1, 2);
+  });
+
+  it('sd fixed', () => {
+    const sd = 50;
+    const height = 3;
+    let data = gaussian.getData({ sd, height });
+    let center = Math.floor((data.length - 1) / 2);
+    expect(data[center]).toBeCloseTo(height, 2);
+    let area = data.reduce((a, b) => a + b, 0);
+    expect(area).toBeCloseTo(height * Math.sqrt(2 * Math.PI) * sd, 2);
+  });
+
+  it('odd fwhm', () => {
+    let data = gaussian.getData({ length: 101, fwhm: 101, height: 1 });
+    expect(data).toHaveLength(101);
+    let lenG = data.length;
+    let center = Math.floor((lenG - 1) / 2);
+    expect(data[center]).toBeCloseTo(1, 4);
+    expect(data[center - 1]).toBeCloseTo(data[center + 1], 4);
+    expect(data[center]).toBeGreaterThan(data[center + 1]);
+  });
+  it('even fwhm', () => {
+    let data = gaussian.getData({ length: 100, fwhm: 100, height: 1 });
+    expect(data).toHaveLength(100);
+    let lenG = data.length;
```

Comment:
We could try to replace the `let` by `const` if it is constant values. There may be a lot of changes for this.